src/main/java/com/spotify/futures/CompletableFutures.java

```diff
@@ -721,6 +725,145 @@ public static void checkCompleted(CompletionStage<?> stage) {
                                                ff.join()));
   }
 
+  /**
+   * Returns the underlying exception of a {@code CompletionException}. This method may be
+   * called on any {@code Throwable} or {@code null}, but if the specified argument is not
+   * an instance of {@code CompletionException} then the argument is returned without
+   * modification.
+   *
+   * <p>This method recursively unpacks multiple layers of {@code CompletionException} wrappers
+   * until it finds a cause that is not a {@code CompletionException}. If there is no such cause,
+   * then the deepest {@code CompletionException} in the chain is returned.
+   *
+   * @param throwable the {@code Throwable} whose underlying cause we want to unravel
+   * @return the actual exception that may or may not be wrapped inside the specified
+   *         {@code Throwable}
+   * @since 0.3.5
+   */
+  public static Throwable unravel(@Nullable Throwable throwable) {
+    if (throwable == null) {
+      return null;
+    }
+    if (!(throwable instanceof CompletionException)) {
+      return throwable;
+    }
+    Throwable cause = throwable.getCause();
+    if (cause == null) {
+      return throwable;
+    }
+    return unravel(cause);
```

Comment:
Is there any risk of getting infinite recursive if cause == self? (Maybe that can never happen for CompletionException)

Reply:
Pretty sure this can't happen. Here's the code for `Throwable.getCause()`:
```
    public synchronized Throwable getCause() {
        return (cause==this ? null : cause);
    }
```